pygsti/circuits/circuit.py

```diff
@@ -4391,7 +4391,10 @@ def done_editing(self):
         """
         if not self._static:
             self._static = True
-            self._labels = tuple([_Label(layer_lbl) for layer_lbl in self._labels])
+            self._labels = tuple([layer_lbl if isinstance(layer_lbl, _Label) 
+                                  else _Label(layer_lbl) for layer_lbl in self._labels])
```

Comment:
Prefer generator to list comprehension. Example:
```python
>>> tuple(a if a > 0 else 0 for a in [1,2,-1,3])
(1, 2, 0, 3)
```

Reply:
I've posted a reply in the corresponding issue thread #450.